tests/e2e/functional_test.go

```diff
@@ -358,7 +358,7 @@ var _ = Describe("Functional e2e", Serial, func() {
 
 		verifyNumaflowControllerReady(Namespace)
 
-		verifyPipelineReady(Namespace, pipelineRolloutName, 2)
+		verifyPipelineReady(Namespace, pipelineRolloutName, 3)
```

Comment:
at this point the PipelineRollout spec has been modified to 3

Reply:
E2E test should be failing without this test?

Reply:
no, first of all the parameter wasn't being used in the function (see other fix below). Plus, the `verifyPodsRunning()` function checks that at least this number are running. And at least 2 were running, so it passed

Reply:
okay, after this comment, decided to update from `>=` to `==`